sympy/series/formal.py

```diff
@@ -1143,6 +1144,192 @@ def integrate(self, x=None, **kwargs):
+        if self.dir != other.dir:
+            raise ValueError("Both series should be calculated from the"
+                             " same direction.")
+        elif self.x0 != other.x0:
+            raise ValueError("Both series should be calculated about the"
+                             " same point.")
+        elif self.x != other.x:
+            raise ValueError("Both series should have the same symbol.")
+
+        f, g = self.function, other.function
+        if other._eval_term(0).as_coeff_mul(other.x)[0] is not S.Zero:
+            raise ValueError("The formal power series of the inner function should not have any "
+                "constant coefficient term.")
+
+        k = Dummy('k')
+        terms = []
+
+        for i in range(1, n):
+            bell_seq = other.coeff_bell(i)
+            fact_seq = sequence(factorial(k), (k, 1, oo))
+            seq = (self.ak * fact_seq * bell_seq)[:i]
+            terms.append(Add(*seq) * self.xk.coeff(i) / factorial(i))
+
+        return self.ind + Add(*terms) + Order(self.xk.coeff(n), (self.x, self.x0))
+
+    def inverse(self, x=None, n=6):
+        r"""
+        Returns the truncated terms of the inverse of the formal power series,
+        up to specified `n`.
+
+        If `f` and `g` are two formal power series of two different functions,
+        then the coefficient sequence ``ak`` of the composed formal power series
+        `fp` will be :-
+
+        .. math::
+        \sum\limits_{k=0}^{n} (-1)^{k} x_0^{-k-1} B_{n,k}(x_1, x_2, \dotsc, x_{n-k+1})
+
+        Parameters
+        ==========
+
+        n : Number, optional
+            Specifies the order of the term up to which the polynomial should
+            be truncated.
+
+        Examples
+        ========
+
+        >>> from sympy import fps, exp, cos, bell
+        >>> from sympy.abc import x
+        >>> f1 = fps(exp(x))
+        >>> f2 = fps(cos(x))
+
+        >>> f1.inverse(x)
+        1 - x + x**2/2 - x**3/6 + x**4/24 - x**5/120 + O(x**6)
+
+        >>> f2.inverse(x, n=8)
+        1 + x**2/2 + 5*x**4/24 + 61*x**6/720 + O(x**8)
+
+        See Also
+        ========
+
+        sympy.functions.combinatorial.numbers.bell
+
+        References
+        ==========
+
+        .. [1] Comtet, Louis: Advanced combinatorics; the art of finite and infinite expansions. Reidel, 1974.
+
+        """
+
+        if x is None:
+            x = self.x
+        if n is None:
+            return iter(self)
+
+        f = self.function
+        if self._eval_term(0) is S.Zero:
+            raise ValueError("Constant coefficient should exist for an inverse of a formal"
+                " power series to exist.")
+        inv = self._eval_term(0)
+
+        k = Dummy('k')
+        terms = []
+
+        for i in range(0, n):
+            bell_seq = self.coeff_bell(i)
+            fact_seq = sequence(factorial(k), (k, 1, oo))
+            sign_seq = sequence((-1, 1), (k, 1, oo))
+            inv_seq = sequence(inv ** (-(k + 1)), (k, 1, oo))
```

Comment:
Can you add these special sequences at time of object creation? You are recreating these every-time.

Reply:
`bell_seq` and `fact_seq` right? I think `sign_seq` and `inv_seq` have been created only once at the time of inversion, i.e, here.